guava/src/com/google/common/math/Quantiles.java

```diff
@@ -513,7 +540,13 @@ private static void checkIndex(int index, int scale) {
    * ({@code required}, {@code to}] are greater than or equal to that value. Therefore, the value at
    * {@code required} is the value which would appear at that index in the sorted dataset.
    */
-  private static void selectInPlace(int required, double[] array, int from, int to) {
+  @SuppressWarnings(value = {"lowerbound:assignment.type.incompatible",/*(1): When entering the loop,
+          required > from and from >= 0, therefore required >= 1.
+          At the assignment, partitionPoint >= required, therefore partitionPoint - 1 >= 0. */
+          "upperbound:assignment.type.incompatible"/*(2): To and from are both < array.length. Entering the loop, from < to,
+          from < array.length - 1, therefore from = partitionPoint + 1 < array.length.
```

Comment:
This doesn't explain why `partitionPoint + 1 < array.length`. Actually, it seems that it doesn't have to be always true. If `from == array.length`, then `from > to` and the loop ends, so the code is still safe.